### PR TITLE
Api flat

### PR DESCRIPTION
**Pull request by fdingiit**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/cel/extract/attributes.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ var Attributemanifest = map[string]attribute.Kind{
 	"api.service":                         attribute.STRING,
 	"api.version":                         attribute.STRING,
 	"api.operation":                       attribute.STRING,
-	"api.protocol":                        attribute.STRING,
+	"api.ProtocolName":                    attribute.STRING,
```
Comment (fdingiit) [inline pkg/cel/extract/attributes.go:79]: 这里改的不对吧

File: pkg/filter/stream/stats/config.go [hunk 2]
```diff
@@ -119,7 +119,7 @@ var defaultDimensions = map[string]string{
 	"destination_service":            `destination.service.host | "unknown"`,
 	"destination_service_name":       `destination.service.name | "unknown"`,
 	"destination_service_namespace":  `destination.service.namespace | "unknown"`,
-	"request_protocol":               `api.protocol | context.protocol | "unknown"`,
+	"request_protocol":               `api.ProtocolName | context.protocol | "unknown"`,
```
Comment (fdingiit) [inline pkg/filter/stream/stats/config.go:122]: 这里改的不对吧

File: go.mod [hunk 3]
```diff
@@ -51,8 +51,12 @@ require (
...
 	k8s.io/klog v1.0.0
-	mosn.io/api v0.0.0-20210129030635-d7dc8206d7b7
+	mosn.io/api v0.0.0-20210127064345-9656ae81df67
 	mosn.io/pkg v0.0.0-20201228090327-daaf86502a50
 )
 
-replace github.com/envoyproxy/go-control-plane => github.com/envoyproxy/go-control-plane v0.9.4
+replace (
+	github.com/envoyproxy/go-control-plane => github.com/envoyproxy/go-control-plane v0.9.4
+	mosn.io/api => github.com/fdingiit/api v0.0.0-20210129094951-d4c2589294e4
```
Comment (cch123) [inline go.mod:60]: 这个不能往 mosn 主仓库提的吧

File: go.mod [hunk 4]
```diff
@@ -51,8 +51,12 @@ require (
...
+	mosn.io/pkg => github.com/fdingiit/pkg v0.0.0-20210129125439-0df41836aba9
```
Comment (cch123) [inline go.mod:61]: 同上

File: pkg/mosn/starter.go [hunk 5]
```diff
@@ -402,6 +407,59 @@ func initializePlugin(log string) {
...
+		switch codec.Type {
+		case v2.GoPlugin:
+			if err := readProtocolPlugin(codec.Path, codec.LoaderFuncName); err != nil {
+				log.StartLogger.Errorf("[mosn] [init codec] init go-plugin codec failed: %+v", err)
+				continue
+			}
+			log.DefaultLogger.Infof("[mosn] [init codec] load go plugin codec succeed: %+v", codec.Path)
+
+		case v2.Wasm:
+		// todo
```
Comment (cch123) [inline pkg/mosn/starter.go:426]: 这里不支持的话，不打个  error 日志么

File: pkg/protocol/types.go [hunk 6]
```diff
@@ -25,11 +25,11 @@ import (
 
 // ProtocolName type definition
 const (
-	Auto      api.Protocol = "Auto"
```
Comment (cch123) [inline pkg/protocol/types.go:28]: 这些似乎理论上不需要改的吧，原来 api 里的类型没被移走吧？

File: go.mod (hunk 3 above)
Comment (nejisama) [reply go.mod:60]: 临时CR使用，符合预期以后再删除replace

**Review by nejisama — APPROVED** [review]

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Pull request merged**